scripts/ci/check_compliance.py

```diff
@@ -1447,6 +1447,62 @@ def run(self):
             with open(file, "r") as fp:
                 self.check_file(file, fp)
 
+class FileEncoding(ComplianceTest):
+    """
+    Check the encoding for source files.
+    """
+    name = "FileEncoding"
+    doc = "Check the encoding for source files."
+    path_hint = "<git-top>"
+
+    MATCH_FILES = [
+            r'.*\.c$',
+            r'.*\.h$',
+            r'.*Kconfig.*',
+            r'.*CMakeLists.txt$',
+            ]
+
+    def match_file(self, file):
+        for match in self.MATCH_FILES:
+            if re.match(match, file):
+                return True
+
+        return False
+
+    def check_file(self, file, fp):
+        if not self.match_file(str(file)):
+            return
+
+        for line_num, line in enumerate(fp.readlines(), start=1):
+            line = line.strip()
+
+            if not line:
+                continue
+
+            if "Copyright" in line:
+                continue
+
+            try:
+                line.encode().decode(encoding="ascii", errors="strict")
```

Comment:
+1 on discouraging the use of emojis (in source code, at least), but things like "µ" should be allowed IMO (or if not, cleanup of existing source might be a good thing to do). So does "ascii" means *extended* ASCII here? (in which case µ would be allowed)?

Reply:
Yeah looking for opinions on exceptions but I'd rather not allow any beyond copyright lines just to avoid maintaining a huge exception list. From what I can tell this is strict ascii, guess printable character, may be worth peeking at the internals of this.

Reply:
Maybe 'latin-1' is worth considering?
It may even be sufficient to cover the copyright line as well.

Reply:
> Maybe 'latin-1' is worth considering?

That would bring in some of the common ones like [°](https://en.wikipedia.org/wiki/%C2%B0) (or is it [º](https://en.wikipedia.org/wiki/%C2%BA)?) [µ](https://en.wikipedia.org/wiki/%CE%9C) [©](https://en.wikipedia.org/wiki/%C2%A9) and [®](https://en.wikipedia.org/wiki/%C2%AE) but also a bunch of other ones that I honestly prefer to never see in stuff like a LOG message